zk/conn.go

```diff
@@ -475,6 +487,17 @@ func (c *Conn) loop() {
 		}
 		c.flushRequests(err)
 
+		if err == ErrSessionExpired && !c.reconnectOnSessionExpiration {
+			atomic.StoreInt32(&c.sessionExpired, 1)
+
+			// The process of flushing requests is racy by design so do it in two steps.
+			c.flushUnsentRequests(err)
```

Comment:
Let me update a diff with reusing logic between Close & session expiration via fixing the race.